security-analysis/security-analysis-api/src/test/java/com/powsybl/security/json/JsonSecurityAnalysisParametersTest.java

```diff
@@ -61,12 +63,61 @@ public void readError() throws IOException {
         }
     }
 
+    @Test
+    public void updateExtensions() {
+        SecurityAnalysisParameters parameters = new SecurityAnalysisParameters();
+        DummyExtension extension = new DummyExtension();
+        extension.setParameterBoolean(false);
+        extension.setParameterString("Bonjour");
+        extension.setParameterDouble(2.8);
+        DummyExtension oldExtension = new DummyExtension(extension);
+        parameters.addExtension(DummyExtension.class, extension);
+        JsonSecurityAnalysisParameters.update(parameters, getClass().getResourceAsStream("/SecurityAnalysisParametersExtensionUpdate.json"));
+        DummyExtension updatedExtension = parameters.getExtension(DummyExtension.class);
+        assertEquals(updatedExtension.isParameterBoolean(), oldExtension.isParameterBoolean());
```

Comment:
Detail for the assertions:
the first parameter should be the "expected" value, so here it would be more logical to invert the 2 parameters (same for other assertions).